great_expectations/core/util.py

```diff
@@ -146,6 +146,7 @@ def convert_to_json_serializable(data):
     if isinstance(data, (SerializableDictDot, SerializableDotDict)):
         return data.to_json_dict()
 
+    # NOTE(cdkini): Deleting this breaks a whole lotta tests
     # Handling "float(nan)" separately is required by Python-3.6 and Pandas-0.23 versions.
     if isinstance(data, float) and np.isnan(data):
         return None
```

Comment:
Made a note here so we could discuss as a team: this comment says this is for 3.6 compatability but deleting the conditional breaks a huge number of tests.

Does anyone have context here? Leaning towards just keeping it to maintain legacy behavior.

Reply:
I just made a PR that should fix this https://github.com/great-expectations/great_expectations/pull/4582